packages/SettingsProvider/src/com/android/providers/settings/SettingsProviderHooks.java

```diff
@@ -0,0 +1,21 @@
+package com.android.providers.settings;
+
+import android.os.UserHandle;
+import android.provider.Settings;
+
+class SettingsProviderHooks {
+
+    static void onSettingsStateInit(final SettingsProvider.SettingsRegistry registry, final int userId) {
+        if (userId == UserHandle.USER_SYSTEM) {
+            SettingsState globalSettings = registry.getSettingsLocked(SettingsProvider.SETTINGS_TYPE_GLOBAL, userId);
+        }
+        SettingsState secureSettings = registry.getSettingsLocked(SettingsProvider.SETTINGS_TYPE_SECURE, userId);
+    }
+
+    private static void insertSetting(SettingsState state, String key, String value) {
+        if (state == null || key == null || value == null) {
```

Comment:
It's better to crash here than to fail silently when `null` is passed.
In general, it's better to add null checks only when `null` is an expected value.